frontend/src/metabase/lib/dom.js

```diff
@@ -26,6 +27,28 @@ export const IFRAMED_IN_SELF = (function() {
   }
 })();
 
+export const WINDOWS_SCROLL_BAR_X_HEIGHT = 20;
+export const MAC_SCROLL_BAR_X_HEIGHT = 10;
+
+// check whether scrollbars are visible to the user,
+// this is off by default on Macs, but can be changed
+// Always on on most other non mobile platforms
+export const areScrollbarsVisible = _.memoize(() => {
+  const scrollableElem = document.createElement("div"),
+    innerElem = document.createElement("div");
+  scrollableElem.style.width = "30px";
+  scrollableElem.style.height = "30px";
+  scrollableElem.style.overflow = "scroll";
+  scrollableElem.style.borderWidth = "0";
+  innerElem.style.width = "30px";
+  innerElem.style.height = "60px";
+  scrollableElem.appendChild(innerElem);
+  document.body.appendChild(scrollableElem); // Elements only have width if they're in the layout
+  const diff = scrollableElem.offsetWidth - scrollableElem.clientWidth;
```

Comment:
Does `diff` reliably give us the scrollbar size?

Reply:
It does!